pkg/controller/storagecluster/storagecluster.go

```diff
@@ -234,7 +235,7 @@ func (c *Controller) Reconcile(ctx context.Context, request reconcile.Request) (
 
 	if err := c.validate(cluster); err != nil {
 		k8s.WarningEvent(c.recorder, cluster, util.FailedValidationReason, err.Error())
-		if updateErr := c.updateStorageClusterState(cluster, corev1.ClusterStateDegraded); updateErr != nil {
+		if updateErr := c.updateLiveStorageClusterState(cluster, corev1.ClusterStateDegraded); updateErr != nil {
```

Comment:
nit: this is actually not needed, as the stc was just fetched a few lines above. https://github.com/libopenstorage/operator/blob/0ecb30b24fe96dcec287d507af4f127554e366a4/pkg/controller/storagecluster/storagecluster.go#L225

I think migration took longer and stc was probably updated somewhere.

Reply:
yes probably this is not needed, most scenarios we don't need this but just incase the cluster is updated somewhere unknown, we won't getting the object update error saying object is updated. And in this function we only update the stc phase, so I think it's good to have.